flink-runtime/src/test/java/org/apache/flink/runtime/resourcemanager/ResourceManagerServiceImplTest.java

```diff
@@ -105,17 +108,15 @@ public void teardown() throws Exception {
             resourceManagerService.close();
         }
 
-        if (leaderElectionService != null) {
-            leaderElectionService.stop();
-        }
-
         if (fatalErrorHandler.hasExceptionOccurred()) {
             fatalErrorHandler.rethrowError();
         }
     }
 
     @AfterClass
     public static void teardownClass() throws Exception {
+        haService.closeAndCleanupAllData();
```

Comment:
This is required because we dont have access to the created `LeaderElection`? Shouldn't this use `TestingLeaderElectionService#triggerContenderCleanup`?

Reply:
tbh I don't know anymore why I added this. The haServices are testing implementations anyway. The LeaderElection will be cleaned up through the `ResourceManagerServiceImpl.close()` call in [line 108](https://github.com/XComp/flink/blob/FLINK-31785/flink-runtime/src/test/java/org/apache/flink/runtime/resourcemanager/ResourceManagerServiceImplTest.java#L108)